File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //get user input for positive value for dollars
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    //convert dollars to cents and round
    int cents = round(dollars * 100);
    //declare values for coins and coin counter
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    int counter = 0;
    //continue whilst there is still cash left
    while (cents > 0)
    {
        //subtract quarter if possible
        if ((cents - quarter) >= 0)
        {
            cents -= quarter;
            counter += 1;
        }
        //otherwise subtract dime if possible
        else if ((cents - dime) >= 0)
        {
            cents -= dime;
            counter += 1;
        }
        //otherwise subtract nickel if possible
        else if ((cents - nickel) >= 0)
        {
            cents -= nickel;
            counter += 1;
        }
        //otherwise subtract a penny
        else
        {
            cents -= penny;
            counter += 1;
        }
    }
    printf("%i\n", counter);

}

File: pset6/cash/cash.py
# Get a positive float as input
amount = 0
while amount <= 0:
    try:
        amount = float(input("Change owed: "))
    except ValueError:
        continue
# convert amount to cents
amount = round(amount, 2) * 100
coin_count = 0
# Try and subtract coins from amount, largest first
while amount > 0:
    # quarter
    if amount >= 25:
        amount -= 25
        coin_count += 1
    # nickel
    elif amount >= 10:
        amount -= 10
        coin_count += 1
    # dime
    elif amount >= 5:
        amount -= 5
        coin_count += 1
    # last must be cent
    else:
        amount -= 1
        coin_count += 1

print(coin_count)
